### PR TITLE
[website] Update the search so you need to submit to trigger a search

### PR DESCRIPTION
**Pull request by loteoo**
For issue: https://github.com/jorgebucaran/hyperapp/issues/933

To search, you now need to submit the search form by hitting "enter".

In mobile, a full-screen form appears when you search.

localhost_1234_search_q=dssdf(iPhone 6_7_8) (1)

### CODE REVIEW

File: docs/src/components/Header/index.jsx [hunk 1]
```diff
@@ -5,12 +5,16 @@ import { Navigate, OpenMenu, CloseMenu } from '../../actions'
 
 const OnSearch = (state, ev) => {
   ev.preventDefault()
-  return [Navigate, `/search?q=${encodeURI(ev.target.value)}`]
+  ev.target.search.select()
+  return [Navigate, `/search?q=${encodeURI(ev.target.search.value)}`]
 }
```
Comment (jorgebucaran) [inline docs/src/components/Header/index.jsx]: This function produces several side effects, that should be wrapped in a proper effect.

File: docs/src/components/Header/index.jsx [hunk 2]
```diff
@@ -5,12 +5,16 @@ import { Navigate, OpenMenu, CloseMenu } from '../../actions'
...
 
-export default ({ menuOpened, location }) => {
-  console.log(location)
+const OnFocus = (state, ev) => {
+  ev.target.select()
+  return state
+}
```
Comment (jorgebucaran) [inline docs/src/components/Header/index.jsx]: This should use a proper effect as well.

**Comment by loteoo** [comment]
@jorgebucaran I've moved the `select()` calls to effects, I'm not sure it's the best implementation tho, do you think it's fine the way I did it?

**Comment by jorgebucaran** [comment]
@loteoo Looks much nicer, thank you! ❤️

**Comment by jorgebucaran** [comment]
@loteoo Looks much nicer, thank you! ❤️ 

Do you want me to merge this now?

**Comment by loteoo** [comment]
@jorgebucaran My pleasure!

Sure, we should be able to close https://github.com/jorgebucaran/hyperapp/issues/933 after merging this one :)

**Comment by jorgebucaran** [comment]
Super! Thank you @loteoo! 💯

**Pull request merged**